### DISCUSSION TITLE
Logging issue with strongswan 5.9.11

### DISCUSSION THREAD

**Discussion by mkr1d** [strongswan/strongswan#1942] [Q&A]
Hi,
I am using Ubuntu 20.04 ARM64 and strongswan ver. 5.9.11
I have an issue with setup of tunnel and I would like to debug the issue. In order to do that ,I want to have as much logging as possible.
I have only limited logging outputs to system.d (journalctl).
I have encountered the file /etc/strongswan.d/charon-logging.conf and have had it configured in the following manner:
charon {

    # Section to define file loggers, see LOGGER CONFIGURATION in
    # strongswan.conf(5).
    filelog {

        # <name> may be the full path to the log file if it only contains
        # characters permitted in section names. Is ignored if path is
        # specified.
        /var/log/charon.log {

            # Loglevel for a specific subsystem.
            # <subsystem> = <default>
	    #cfg = 2
            #ike = 2
            # If this option is enabled log entries are appended to the existing
            # file.
             append = yes

            # Default loglevel.
             default = 2

            # Enabling this option disables block buffering and enables line
            # buffering.
             flush_line = yes

            # Prefix each log entry with the connection name and a unique
            # numerical identifier for each IKE_SA.
            # ike_name = no

            # Optional path to the log file. Overrides the section name. Must be
            # used if the path contains characters that aren't allowed in
            # section names.
            # path = /var/log/charon.log

            # Adds the milliseconds within the current second after the
            # timestamp (separated by a dot, so time_format should end with %S
            # or %T).
             time_add_ms = yes

            # Prefix each log entry with a timestamp. The option accepts a
            # format string as passed to strftime(3).
             time_format =  %b %e %T

         }

    }

    # Section to define syslog loggers, see LOGGER CONFIGURATION in
    # strongswan.conf(5).
    syslog {

        # Identifier for use with openlog(3).
        # identifier =

        # <facility> is one of the supported syslog facilities, see LOGGER
        # CONFIGURATION in strongswan.conf(5).
        # <facility> {

            # Loglevel for a specific subsystem.
            # <subsystem> = <default>

            # Default loglevel.
            # default = 1

            # Prefix each log entry with the connection name and a unique
            # numerical identifier for each IKE_SA.
            # ike_name = no

        # }

    }

}

The lines with cfg,ike and path I have tried to toggle on/off with no impact.
The result, as far as I understand ,ought to be at the very least creation of the file /var/log/charon.log and of course ,writing of log into it.Unfortunately ,this is not the result and I don't see the files and/or log content of charon.
Also ,I have tried to set the same information on strongswan.conf,but ipsec wasn't able to run because of configuration error,which makes sense.
Please advise.
Thank You!

**Comment by tobiasbrunner** [comment]
Have you read [the docs about logging](https://docs.strongswan.org/docs/5.9/config/logging.html) and the recommendations regarding it in #196?

> I have encountered the file /etc/strongswan.d/charon-logging.conf and have had it configured in the following manner:

That only has an effect if it's included in `/etc/strongswan.conf`.




> The result, as far as I understand ,ought to be at the very least creation of the file /var/log/charon.log and of course ,writing of log into it

It's possible the IKE daemon doesn't have permission to create that file if it's prevented by e.g. AppArmor.

> I have tried to set the same information on strongswan.conf,but ipsec wasn't able to run because of configuration error,which makes sense.

What do you mean? Why would a configuration error make sense?
